pkg/controller/controller.go

```diff
@@ -434,17 +441,26 @@ func (c *Controller) runAction(ctx context.Context, t *tomb.Tomb, as *crv1alpha1
 	c.logAndSuccessEvent(ctx, fmt.Sprintf("Executing action %s", action.Name), "Started Action", as)
 	tp, err := param.New(ctx, c.clientset, c.dynClient, c.crClient, c.osClient, action)
 	if err != nil {
+		if c.metrics != nil {
+			c.metrics.actionSetResolutionCounterVec.WithLabelValues(ACTION_SET_COUNTER_VEC_LABEL_RES_FAILURE).Inc()
+		}
```

Comment:
Would it be a good idea to extract this into a utility function

```
func (c) incActionsetResConuterVec(success||failure){
if c.metrics != nil {
			c.metrics.actionSetResolutionCounterVec.WithLabelValues(success||failure).Inc()
		}
}
```

and then the call can just be

```
tp, err := param.New(ctx, c.clientset, c.dynClient, c.crClient, c.osClient, action)
	if err != nil {
	        c.incActionsetResConuterVec(success||failure)
               return err
        }
```

Reply:
Sure, this makes sense. Done